tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/IResourceChangeListenerTest.java

```diff
@@ -529,13 +529,16 @@ public void testTouchFileWithAutobuildOff() throws Exception {
 		final IWorkspace workspace = getWorkspace();
 		try {
 			setAutoBuilding(false);
+			// make sure the events do not get fired from autobuild:
+			((Workspace) getWorkspace()).getBuildManager().waitForAutoBuild();
```

Comment:
I believe after aufobuild was switched off, we should wait for both jobs types, not just for one.
So why not join to the job family?

Reply:
doesnt matter if the "off" job runs or not. it's only important here that the autobuild is not running anymore,  because the listener can not distinguish who called him.